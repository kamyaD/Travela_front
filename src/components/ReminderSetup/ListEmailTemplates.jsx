import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import WithLoadingEmailTemplatesTable from './EmailTemplatesTable';
import TemplatesPagination from './TemplatesPagination';
import NoTemplates from './NoTemplates';

class ListEmailTemplates extends Component {
  constructor(props){
    super(props);
  }
  componentDidMount() {
    const { fetchTemplates, location: {search} } = this.props;
    fetchTemplates(search);
  }

  onPageChange = (previousOrNext) => {
    const { fetchTemplates, listEmailTemplatesReducer:{pagination: { currentPage }}, location: {search} } = this.props;
    fetchTemplates(`${search}${search?'&':'?'}page=${previousOrNext === 'previous'?currentPage-1:currentPage+1}`);
  };


  render(){
    const { listEmailTemplatesReducer: {templates, pagination } } = this.props;
    const { pageCount, currentPage } = pagination;
    return (
      <Fragment>
        {templates.length>0?(
          <div>
            <WithLoadingEmailTemplatesTable
              templates={templates}
            />
            <TemplatesPagination
              onPageChange={this.onPageChange}
              pageCount={pageCount?pageCount:1}
              currentPage={currentPage?currentPage:1}
            />
          </div>
        ):
          (<NoTemplates />)}
      </Fragment>
    );
  }
}

ListEmailTemplates.propTypes = {
  fetchTemplates: PropTypes.func.isRequired,
  listEmailTemplatesReducer: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired
};

export default ListEmailTemplates;
