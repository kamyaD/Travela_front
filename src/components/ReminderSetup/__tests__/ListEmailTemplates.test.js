import React from 'react';
import { mount } from 'enzyme';
import ListEmailTemplates from '../ListEmailTemplates';
import listOfTemplates from '../../../views/ReminderSetup/__mocks__/index';

describe('<ListEmailTemplates />', () => {
  const {templates, pagination} = listOfTemplates;
  const props = {
    fetchTemplates: jest.fn(),
    setItemToDisable: jest.fn(),
    disableEnable: false,
    listEmailTemplatesReducer: {
      pagination: {
        pageCount: 3,
        currentPage: 3
      },
      templates: [{
        id: 1,
        name: '',
        creator : {
          fullName: ''
        },
        createdAt: '2019-01-21T07:33:26.123Z',
        updatedAt: '2019-01-21T07:33:26.123Z',
        deletedAt: null,
        disabled: true,
      }]
    },
    location: {}
  };

  const wrapper =  mount(<ListEmailTemplates {...props} />);

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onPageChange function', () => {
    
    wrapper.find('#next-button').simulate('click');
    expect(props.fetchTemplates).toHaveBeenCalled();
  });
  
});
 
