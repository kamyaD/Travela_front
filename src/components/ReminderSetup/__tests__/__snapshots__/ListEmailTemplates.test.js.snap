// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListEmailTemplates /> renders without crashing 1`] = `
<div>
  <div
    className="list-templates"
  >
    <div
      className="table__container"
    >
      <table
        className="mdl-data-table mdl-js-data-table readiness-table"
      >
        <thead>
          <tr>
            <th
              className="mdl-data-table__cell--non-numeric table__head"
            >
              Template Name
            </th>
            <th
              className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
            >
              <span />
            </th>
            <th
              className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
            >
              Created By
            </th>
            <th
              className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
            >
              <span />
            </th>
            <th
              className="mdl-data-table__cell--non-numeric table__head"
            >
              Created On
            </th>
            <th
              className="mdl-data-table__cell--non-numeric table__head"
            />
          </tr>
        </thead>
        <tbody
          className="table__body"
        >
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              testing2
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              21-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              moses branch testing
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              21-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              Passport Email Template
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              18-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              Passport Email Template
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              18-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              Passport Email Template
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              18-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
          <tr
            className="table__rows "
          >
            <td
              className="mdl-data-table__cell--non-numeric table__data readiness__cell-name"
            >
              Visa renewal
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              Kevin Koech
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              <span />
            </td>
            <td
              className="mdl-data-table__cell--non-numeric table__data"
            >
              18-01-2019
            </td>
            <td
              className="table__data"
            >
              <span>
                <i
                  className="fa fa-ellipsis-v on"
                  id="toggleIcon"
                  onClick={[Function]}
                  role="presentation"
                />
                <div
                  className="table__menu-container "
                >
                  <ul
                    className="table__menu-list"
                  >
                    <li
                      className="table__menu-list-item top"
                    >
                      <span
                        className="edit"
                      >
                        Edit
                      </span>
                    </li>
                    <li
                      className="table__menu-list-item bottom"
                    >
                      <span
                        className="disable"
                      >
                        Disable
                      </span>
                    </li>
                  </ul>
                </div>
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="reminder-email-templates"
  >
    <div
      className="pagination"
    >
      <button
        className="pagination__button"
        disabled={true}
        id="previous-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className=""
        >
          Previous
        </span>
        <span
          className=" mdl-cell--hide-desktop mdl-cell--hide-tablet"
        >
          Previous
        </span>
      </button>
      <div
        className="pagination__items"
      >
        <span
          className="pagination__page"
        >
          Page
        </span>
        <div
          className="pagination__current-page"
          id="current-page"
        >
          1
        </div>
        of
        <span
          className="pagination__all-pages"
          id="page-count"
        >
          5
        </span>
      </div>
      <button
        className="pagination__button"
        disabled={false}
        id="next-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className=""
        >
          Next
        </span>
        <span
          className=" mdl-cell--hide-desktop mdl-cell--hide-tablet"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</div>
`;
